File: app/src/main/java/com/lazycouple/restapiclient/ui/viewModel/AdapterViewModel.java
package com.lazycouple.restapiclient.ui.viewModel;

import android.databinding.BaseObservable;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by amuyu on 2017. 6. 12..
 */

public abstract class AdapterViewModel<T> extends BaseObservable {
    List<T> lists = new ArrayList<T>();

    public List<T> getLists() {
        return lists;
    }

    public void setLists(List<T> lists) {
        this.lists = lists;
    }

    public void addItem(T item) {
        lists.add(item);
    }

    public int getItemCount() {
        return lists.size();
    }

    public long getItemId(int position) {
        return lists.get(position).hashCode();
    }

    public T getItem(int position) {
        return lists.get(position);
    }

    public void clearLists() {
        lists.clear();
    }
}

File: app/src/main/kotlin/com/lazycouple/restapiclient/ui/viewModel/AdapterViewModel.kt
package com.lazycouple.restapiclient.ui.viewModel

import com.lazycouple.restapiclient.util.ObservableViewModel

/**
 * Created by amuyu on 2017. 6. 12..
 */

abstract class AdapterViewModel<T> : ObservableViewModel() {
    internal var lists: ArrayList<T> = ArrayList()

    fun getLists(): List<T> {
        return lists
    }

    fun addItem(item: T) {
        lists.add(item)
    }

    val itemCount: Int
        get() = lists.size

    fun getItemId(position: Int): Long {
        return lists[position]?.hashCode()?.toLong() ?: -1
    }

    fun getItem(position: Int): T {
        return lists[position]
    }

    fun clearLists() {
        lists.clear()
    }
}
